src/perf/tracepoint/format.cpp

```diff
@@ -30,32 +30,74 @@
 #include <boost/filesystem.hpp>
 #include <boost/filesystem/fstream.hpp>
 
+#include <cstdio>
+
 namespace lo2s
 {
 namespace perf
 {
 namespace tracepoint
 {
+void EventFormat::parse_id()
+{
+    // Yes, this is C file I/O.  Tell me when you have found a way to easily
+    // access POSIX filesystem errors in C++.
+
+    boost::filesystem::path id_path{ base_path_ / name_ / "id" };
+
+    std::FILE* id_file = std::fopen(id_path.c_str(), "rb");
```

Comment:
I have the feeling, this error handling should be a separate function, so we can reuse it.

Reply:
It may become a bit nifty in collecting / logging the information from the different layers properly, but I agree that it is a good idea.